### ISSUE TITLE
训练出错

### ISSUE THREAD

**Issue by Tianny109** [zysxmu/DDTB#5]
您好！我在尝试将您的代码应用到EDSR_full时遇到了如下问题：
我按照run.sh中使用如下命令：
`python main_limitrange_incremental.py --scale 4 --k_bits 2 --model EDSR --pre_train ../pretrained_model/edsr_x4.pt --patch_size 192 --data_test Set14 --dynamic_ratio 0.3 --save "output/edsrx4/2bit" --dir_data ../../daq/datasets --print_every 100 --n_resblocks 32 --n_feats 256 --res_scale 0.1`
但是出现nonloss：
`142 1678
forcheck 352
dynamic_ratio 0.3
output/edsrx4/2bit start!
[Epoch 1]       Learning rate: 1.00e-4
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
Calibration! Accumulate the max_v and min_v for computing Variance
/root/miniconda3/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:131: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "

Evaluation:
0 96
100%|███████████████████████████████████████████| 14/14 [00:02<00:00,  4.82it/s]
[Set14 x4] PSNR: 28.815  (Best: 28.815 @epoch 1)
save: ../experiment/output/edsrx4/2bit/model/checkpoint.pth.tar
Total: 3.26s

352
[('body.31.quant_act2', 111096.305), ('body.30.quant_act2', 39680.836), ('body.29.quant_act2', 20181.018), ('body.28.quant_act2', 11736.804), ('body.27.quant_act2', 7554.008), ('body.31.quant_act1', 6049.1787), ('body.24.quant_act2', 6014.326), ('body.26.quant_act2', 5575.9478), ('body.30.quant_act1', 4959.7163), ('body.31.quant_act3', 4544.2764), ('body.1.quant_act2', 4183.447), ('body.25.quant_act2', 4142.0376), ('body.0.quant_act2', 4085.177), ('body.29.quant_act1', 3485.3975), ('body.23.quant_act2', 3320.9275), ('body.2.quant_act2', 3032.9412), ('body.28.quant_act1', 2772.5986), ('body.21.quant_act2', 2526.322), ('body.22.quant_act2', 2492.3462), ('body.27.quant_act1', 2414.1802), ('body.3.quant_act2', 2250.6243), ('body.20.quant_act2', 2213.8235), ('body.26.quant_act1', 2198.696), ('body.25.quant_act1', 2126.0864), ('body.24.quant_act1', 2039.301), ('body.23.quant_act1', 1976.9053), ('body.22.quant_act1', 1921.5474), ('body.21.quant_act1', 1868.2605), ('body.20.quant_act1', 1840.9009), ('body.19.quant_act1', 1823.9797), ('body.19.quant_act2', 1791.7413), ('body.18.quant_act1', 1783.1492), ('body.17.quant_act1', 1757.79), ('body.16.quant_act1', 1723.9619), ('body.15.quant_act1', 1691.513), ('body.14.quant_act1', 1653.3838), ('body.13.quant_act1', 1602.7052), ('body.12.quant_act1', 1592.1611), ('body.11.quant_act1', 1575.512), ('body.9.quant_act1', 1563.3276), ('body.6.quant_act1', 1560.6414), ('body.7.quant_act1', 1559.1096), ('body.5.quant_act1', 1556.2207), ('body.10.quant_act1', 1554.7053), ('body.4.quant_act2', 1550.6201), ('body.4.quant_act1', 1544.1685), ('body.8.quant_act1', 1543.9205), ('body.3.quant_act1', 1534.7623), ('body.18.quant_act2', 1518.69), ('body.17.quant_act2', 1458.9325), ('body.15.quant_act2', 1330.0747), ('body.2.quant_act1', 1313.9873), ('body.10.quant_act2', 1278.9199), ('body.11.quant_act2', 1276.8789), ('body.14.quant_act2', 1171.3179), ('body.13.quant_act2', 1165.5875), ('body.6.quant_act2', 1152.7964), ('body.30.quant_act3', 1079.5452), ('body.5.quant_act2', 1013.2062), ('body.16.quant_act2', 997.2145), ('body.0.quant_act1', 982.44836), ('body.12.quant_act2', 904.49176), ('body.9.quant_act2', 875.8664), ('body.8.quant_act2', 868.754), ('body.1.quant_act1', 794.1039), ('body.7.quant_act2', 734.4374), ('body.29.quant_act3', 658.3931), ('body.1.quant_act3', 359.22583), ('body.0.quant_act3', 281.16086), ('body.28.quant_act3', 252.90512), ('body.27.quant_act3', 166.35794), ('body.26.quant_act3', 108.88365), ('body.24.quant_act3', 107.767715), ('body.25.quant_act3', 96.11083), ('body.2.quant_act3', 86.63263), ('body.23.quant_act3', 59.232838), ('body.22.quant_act3', 41.016792), ('body.3.quant_act3', 37.784084), ('body.5.quant_act3', 35.57997), ('body.21.quant_act3', 34.89799), ('body.20.quant_act3', 31.723759), ('body.19.quant_act3', 26.107613), ('body.13.quant_act3', 22.593678), ('body.15.quant_act3', 22.41423), ('body.14.quant_act3', 21.250359), ('body.18.quant_act3', 20.088255), ('body.11.quant_act3', 18.722145), ('body.6.quant_act3', 18.017832), ('body.4.quant_act3', 16.826809), ('body.17.quant_act3', 16.078762), ('body.16.quant_act3', 15.053723), ('body.12.quant_act3', 14.488768), ('body.10.quant_act3', 14.429821), ('body.7.quant_act3', 13.885572), ('body.9.quant_act3', 11.664296), ('body.8.quant_act3', 10.156562)]
dynamice_names {'body.3.quant_act2', 'body.28.quant_act2', 'body.0.quant_act2', 'body.28.quant_act1', 'body.21.quant_act2', 'body.25.quant_act2', 'body.26.quant_act2', 'body.30.quant_act2', 'body.26.quant_act1', 'body.31.quant_act3', 'body.27.quant_act1', 'body.25.quant_act1', 'body.27.quant_act2', 'body.20.quant_act2', 'body.31.quant_act2', 'body.2.quant_act2', 'body.22.quant_act1', 'body.24.quant_act2', 'body.24.quant_act1', 'body.23.quant_act1', 'body.29.quant_act1', 'body.21.quant_act1', 'body.22.quant_act2', 'body.29.quant_act2', 'body.1.quant_act2', 'body.31.quant_act1', 'body.23.quant_act2', 'body.30.quant_act1'}
len dynamice_names, 28 len var_list, 96
[Epoch 2]       Learning rate: 1.00e-4
clibration!
...
clibration!
**None loss!!**
Python 3.8.10 (default, Jun  4 2021, 15:09:15) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.27.0 -- An enhanced Interactive Python. Type '?' for help.`
不知道是什么原因 望您指点

**Comment by zysxmu** [comment]
EDSR不应该会出现这个问题，要检查一下获得的l量化参数的范围

**Comment by Tianny109** [comment]
感谢您的建议 我现在已经可以正常训练了

**Comment by zysxmu** [comment]
ok, 欢迎进一步交流
